vocode/streaming/synthesizer/eleven_labs_synthesizer.py

```diff
@@ -34,29 +36,47 @@ def __init__(
         self.similarity_boost = config.similarity_boost
         self.words_per_minute = 150
 
-    def create_speech(
+    async def create_speech(
         self,
         message: BaseMessage,
         chunk_size: int,
         bot_sentiment: Optional[BotSentiment] = None,
     ) -> SynthesisResult:
-        self.elevenlabs.set_api_key(self.api_key)
         voice = self.elevenlabs.Voice(voice_id=self.voice_id)
         if self.stability is not None and self.similarity_boost is not None:
             voice.settings = self.elevenlabs.VoiceSettings(
                 stability=self.stability, similarity_boost=self.similarity_boost
             )
+        url = ELEVEN_LABS_BASE_URL + f"text-to-speech/{self.voice_id}"
+        headers = {"xi-api-key": self.api_key}
+        body = {
+            "text": message.text,
+            "voice_settings": voice.settings.dict() if voice.settings else None,
+        }
 
-        audio = self.elevenlabs.generate(message.text, voice=voice)
+        async with aiohttp.ClientSession() as session:
```

Comment:
since we've decoupled the final transcription worker, agent response worker, and synthesis result worker, this gives us parallel synthesis right? since the SynthesisResultsWorker is playing while the AgentResponseWorker is synthesizing

Reply:
The synthesis runs in parallel of the other worker types, but there is only sentence / message at most being synthesized at a given time. I have started working on a AsyncWorker class that can allow some level of concurrency within one worker type. The default will be 2.

I realize that the wording is a bit confusing here because parallelization is dependent on the context. I think it would be clearer to describe your this change "decoupling the synthesizer using AsyncWorker" and reserving "parallelization" for the upcoming ConcurrentAsyncWorker

Reply:
right, fair enough.